params/network_upgrades.go

```diff
@@ -64,16 +66,40 @@ func (m *MandatoryNetworkUpgrades) mandatoryForkOrder() []fork {
 	}
 }
 
-// OptionalNetworkUpgrades includes overridable and optional Subnet-EVM network upgrades.
-// These can be specified in genesis and upgrade configs.
-// Timestamps can be different for each subnet network.
-// TODO: once we add the first optional upgrade here, we should uncomment TestVMUpgradeBytesOptionalNetworkUpgrades
-type OptionalNetworkUpgrades struct{}
+type OptionalFork struct {
+	Block     *big.Int `json:"block" serialize:"true"`
+	Timestamp *uint64  `json:"timestamp" serialize:"true"`
+}
+
+func (m *OptionalFork) ToFork(name string) fork {
+	var block *big.Int
+	var timestamp *uint64
+	if m.Block != nil {
+		block = m.Block
+	}
+	if m.Timestamp != nil {
+		timestamp = m.Timestamp
+	}
+	return fork{
+		name:      name,
+		block:     block,
+		timestamp: timestamp,
+		optional:  true,
+	}
+}
+
+type OptionalNetworkUpgrades struct {
+	Test *OptionalFork `json:"test,omitempty" serialize:"true,nullable"`
+}
 
 func (n *OptionalNetworkUpgrades) CheckOptionalCompatible(newcfg *OptionalNetworkUpgrades, time uint64) *ConfigCompatError {
 	return nil
 }
 
 func (n *OptionalNetworkUpgrades) optionalForkOrder() []fork {
```

Comment:
can we use the same logic as we do it in `mandatoryForkOrder`

Reply:
The expected output of `optionalForkOrder` is the same as `mandatoryForkOrder`. However, to make things less error-prone `OptionalFork` has a method `ToFork` that will return the expected struct.